### PR TITLE
[YANG] Add trap flow counter to yang model

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### Why I did it

Add trap flow counter to yang model

#### How I did it

Add a new container for yang model and add unit test config

#### How to verify it

Build

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

**Comment by dgsudharsan** [comment]
@praveen-li Can you please sign off?

**Comment by dgsudharsan** [comment]
@praveen-li  @qiluo-msft  Can you please review?

File: src/sonic-yang-models/yang-models/sonic-flex_counter.yang [hunk 1]
```diff
@@ -161,6 +161,16 @@ module sonic-flex_counter {
                 }
             }
 
+            container FLOW_CNT_TRAP {
```
Comment (praveen-li) [inline src/sonic-yang-models/yang-models/sonic-flex_counter.yang:164]: I feel, We can use full name FLOW_COUNTER_TRAP

**Review by praveen-li — APPROVED** [review]
LGTM, name change is a suggestion, but I am not sure if backend code is already merged.

File: src/sonic-yang-models/yang-models/sonic-flex_counter.yang (hunk 1 above)
Comment (Junchao-Mellanox) [reply src/sonic-yang-models/yang-models/sonic-flex_counter.yang:164]: Thanks for the comment. Backend code has been merged, so will keep this name for now.

**Review by prsunny — APPROVED** [review]

**Comment by judyjoseph** [comment]
Please raise a new PR for 202111 branch, as this patch cannot be cleanly cherry-picked.

**Comment by Junchao-Mellanox** [comment]
This change is already in 202111.

**Pull request merged**